### DISCUSSION TITLE
Preventing scroll to top not supported by `Router.push`?

### DISCUSSION THREAD

**Discussion by oyeanuj** [vercel/next.js#14534] [Help]
Hi everyone, not sure if this is a bug report or intended behavior but I noticed that if you use `Router.push`, there is no way to mention the prevention of scroll to top, like there is for `Link`.  Here are the docs for this: https://nextjs.org/docs/api-reference/next/router#routerpush

It seems like here in the code, there needs to be a check for `scroll`: 

https://github.com/vercel/next.js/blob/23db226ea5721e5f4bc14e3b79efa03143e0657f/packages/next/next-server/lib/router/router.ts#L738-L740

@Timer @timneutkens Is there a hidden way to do this and docs just need to be updated or there is a reason for `router.push` to not support `scroll`?

@everyone-else: Do you have workarounds in the meantime?

**Comment by Bleuzen** [comment]
Also when using the browsers navigation buttons (going back) it scrolls to the top. Searching for a way to prevent this...

**Comment by oyeanuj** [comment]
@Bleuzen Since there wasn't an answer here and thinking more about it, I ended up creating a feature request ticket here: https://github.com/vercel/next.js/issues/15206 

Feel free to add your use-case more in detail.

**Comment by Manikanta-20** [comment]


**Reply by Bleuzen** [reply]
This does the opposite of preventing the scroll. The goal here is to NOT scroll to top on route change.
